org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleBuildSupport.java

```diff
@@ -49,6 +56,23 @@ public void update(IProject project, IProgressMonitor monitor) throws CoreExcept
 	public boolean isBuildFile(IResource resource) {
 		return resource != null
 				&& resource.getType()== IResource.FILE
-				&& resource.getName().endsWith(".gradle");
+				&& resource.getName().endsWith(GRADLE_SUFFIX);
+	}
+
+	public static void cleanGradleModel(IProgressMonitor monitor) {
```

Comment:
cleanGradleModels

you should add some javadoc explaining stale gradle project preferences are being deleted